api/desecapi/serializers.py

```diff
@@ -645,7 +640,12 @@ class Meta:
         extra_kwargs = UserSerializer.Meta.extra_kwargs
 
     def validate_domain(self, value):
-        DomainSerializer.raise_if_domain_unavailable(value, self.context['request'].user)
+        serializer = DomainSerializer(data=dict(name=value), context=self.context)
+        try:
+            serializer.is_valid(raise_exception=True)
+        except serializers.ValidationError:
+            raise serializers.ValidationError(serializer.default_error_messages['name_unavailable'],
```

Comment:
If `DomainSerializer` is ever extended to check other things as well, this will behave strangely. Also, why catch the exception and re-raise instead of just letting it bubble up?

Reply:
I'm not sure what you mean, but if that really happens, we'll notice thanks to our tests.

The exception is re-raised because otherwise, the JSON error output will be nested like `{"domain": {"name": "..."}}` although the input field of that endpoint is just a string with name `domain`.

Reply:
I meant: what if `DomainSerializer` is extended to have another mandatory field? The code that you propose here does the job, because currently the only validation of `DomainSerializer` is validity and available of the name. So I wanted to point out that want we want here is that the domain *name* is valid, but what we are checking is that `{name: <domainname>}` is a valid serialization for the `DomainSerializer`. These two things currently coincide, but this may change in the future.

Reply:
Got it. Previously, we took the other approach (`raise_if_name_unavailable()` or something), but that caused a potential inconsistency: `DomainSerializer` has an extra `trim_whitespace` setting which the `raise_if_name_unavailable()` code was not aware of. I felt that it would be best to use the same code path for any domain name validation.

Reply:
My feeling is that the proper way forward with this would be to create a subclass of [CharField](https://www.django-rest-framework.org/api-guide/fields/#charfield) that checks if the given string is a valid and available domain name. This could then be used here and in `DomainSerializer`. I guess that's nice to have, not a requirement for this PR